warehouse/internal/repo/table_upload.go

```diff
@@ -0,0 +1,375 @@
+	if err != nil {
+		return model.TableUpload{}, fmt.Errorf("querying table uploads: %w", err)
+	}
+
+	entries, err := repo.parseRows(rows)
+	if err != nil {
+		return model.TableUpload{}, fmt.Errorf("parsing rows: %w", err)
+	}
+	if len(entries) == 0 {
+		return model.TableUpload{}, fmt.Errorf("no table upload found with uploadID: %d, tableName: %s", uploadID, tableName)
+	}
+
+	return entries[0], err
+}
+
+func (*TableUploads) parseRows(rows *sql.Rows) ([]model.TableUpload, error) {
+	var tableUploads []model.TableUpload
+
+	defer func() { _ = rows.Close() }()
+
+	for rows.Next() {
+		var (
+			tableUpload     model.TableUpload
+			locationRaw     sql.NullString
+			lastExecTimeRaw sql.NullTime
+			totalEvents     sql.NullInt64
+		)
+		err := rows.Scan(
+			&tableUpload.ID,
+			&tableUpload.UploadID,
+			&tableUpload.TableName,
+			&tableUpload.Status,
+			&tableUpload.Error,
+			&lastExecTimeRaw,
+			&totalEvents,
+			&tableUpload.CreatedAt,
+			&tableUpload.UpdatedAt,
+			&locationRaw,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("scanning row: %w", err)
+		}
+
+		tableUpload.CreatedAt = tableUpload.CreatedAt.UTC()
+		tableUpload.UpdatedAt = tableUpload.UpdatedAt.UTC()
+
+		if lastExecTimeRaw.Valid {
+			tableUpload.LastExecTime = lastExecTimeRaw.Time.UTC()
+		}
+		if locationRaw.Valid {
+			tableUpload.Location = locationRaw.String
+		}
+		if totalEvents.Valid {
+			tableUpload.TotalEvents = totalEvents.Int64
+		}
+
+		tableUploads = append(tableUploads, tableUpload)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("iterating rows: %w", err)
+	}
+
+	return tableUploads, nil
+}
+
+func (repo *TableUploads) PopulateTotalEventsFromStagingFileIDs(ctx context.Context, uploadId int64, tableName string, stagingFileIDs []int64) error {
+	subQuery := `
+		WITH row_numbered_load_files as (
+		  SELECT
+			total_events,
+			row_number() OVER (
+			  PARTITION BY staging_file_id,
+			  table_name
+			  ORDER BY
+				id DESC
+			) AS row_number
+		  FROM
+			` + loadTableName + `
+		  WHERE
+			staging_file_id = ANY($3)
+			AND table_name = $2
+		)
+		SELECT
+		  sum(total_events) as total
+		FROM
+		  row_numbered_load_files
+		WHERE
+		  row_number = 1
+`
+	query := `
+		UPDATE
+			` + tableUploadTableName + `
+		SET
+		  total_events = subquery.total
+		FROM
+		  (` + subQuery + `) AS subquery
+		WHERE
+		  wh_upload_id = $1 AND
+		  table_name = $2;
+`
+	queryArgs := []any{
+		uploadId,
+		tableName,
+		pq.Array(stagingFileIDs),
+	}
+	result, err := repo.db.ExecContext(
+		ctx,
+		query,
+		queryArgs...,
+	)
+	if err != nil {
+		return fmt.Errorf(`set total events: %w`, err)
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf(`rows affected: %w`, err)
+	}
+	if rowsAffected == 0 {
+		return fmt.Errorf(`no rows affected`)
+	}
+
+	return nil
+}
+
+func (repo *TableUploads) GetSumOfTotalExportedEventsForUploadID(ctx context.Context, uploadId int64, skipTables []string) (int64, error) {
```

Comment:
[minor] simplify name
```suggestion
func (repo *TableUploads) TotalExportedEvents(ctx context.Context, uploadId int64, skipTables []string) (int64, error) {
```